mesonbuild/backend/backends.py

```diff
@@ -355,6 +355,10 @@ def get_target_filename_for_linking(self, target: T.Union[build.Target, build.Cu
         # On all other platforms, we link to the library directly.
         if isinstance(target, build.SharedLibrary):
             link_lib = target.get_import_filename() or target.get_filename()
+            # In AIX, if we archive .so, the blibpath must link to archived shared library otherwise to the .so file.
+            if mesonlib.is_aix():
+                if target.aix_so_archive is True:
```

Comment:
`if target.aix_so_archive:`

(You don't need "is True" here.)

I would also put it all on a single line:
```python
if mesonlib.is_aix() and target.aix_so_archive:
```

Reply:
Done